core/evm.go

```diff
@@ -44,6 +44,14 @@ func NewEVMContext(msg Message, header *types.Header, chain ChainContext, author
 	} else {
 		beneficiary = *author
 	}
+	var paygasMode = vm.PaygasNoOp
+	if msg.IsAA() {
+		if msg.To() != nil {
```

Comment:
hmmm, why does msg.To() == nil turn on PaygasHalt mode? this seems generally hacky.

Reply:
Good point. I removed the distinction between `PaygasContinue` and `PaygasHalt` from `NewEVMContext`.
I now always set it to `PaygasContinue` there and then to `PaygasHalt` during `Validate()`